src/rust/engine/fs/store/src/local.rs

```diff
@@ -269,8 +269,6 @@ impl ByteStore {
   /// blocking, this accepts a function that views a slice rather than returning a clone of the
   /// data. The upshot is that the database is able to provide slices directly into shared memory.
   ///
-  /// The provided function is guaranteed to be called in a context where it is safe to block.
```

Comment:
Have you audited all of the other transitive callers to make sure they don't block? IIRC a bunch of things like `materialize_file` end up calling through here too?

Reply:
Yep, we do use this for a couple places like `materialize_file`. 

Maybe I was wrong to remove this comment? It is _safe_ to block, only it is not _performant_. But I spent a few minutes checking all the transitive call sites and I do think that we want to block on all those things because they are synchronous operations like materializing a digest.

Reply:
Yeah, I'd restore the comment if it's still valid :) Thanks for checking!